### PR TITLE
chore(table): update column management demo with new data

### LINKED ISSUE #7991: Table: Update column management demo data

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #7991 

Bonus updates!!:
- Cleaned up the demo! Now 500 lines shorter/cleaner 🧼
- Converted to function component with hooks 🪝
- Converted to use `TableComposable` which is the PF recommendation over legacy table 
- Moved demo to standalone file which is the convention we are following for other PF components/demos

**Demo link for convenience:** https://patternfly-react-pr-8044.surge.sh/components/table/react-demos/column-management/

### CODE REVIEW

**Comment by mcarrano** [comment]
@jenny-s51 This looks good to me. But I'll defer to @mmenestr as I believe she worked on the original design.

**Comment by mmenestr** [comment]
This looks good as far as the manage column function goes! I would just add a Page header to the page because it looks a bit weird to just have breadcrumbs

**Review by jenny-s51 — COMMENTED** [review]
Thanks @mmenestr , nice catch! Updated the demo to include a main section.

**Review by nicolethoen — COMMENTED** [review]
This looks great! Wow!

It may benefit consumers to include a few more comments within the demo code to guide a consumer through the features/considerations made to implement the various bits. I found that when I read 'setFilteredRows' I began looking for the implementation of the 'Name' select filter. But that's not what 'setFilteredRows' is referring to. So, clarity, even just with additional comments, might be helpful in the `fitlerData` function - and maybe also to clearly separate the pagination logic from the column management logic in the code?

I'm curious if there should be a way to manage the columns in the mobile/smaller viewport screen widths? I'm not sure where the 'manage columns' button would go.

**Comment by nicolethoen** [comment]
I also spent some time wondering if there was a better way to control column widths so that the widths dont jump around as you paginate. I'm not sure, because since the columns obviously change as they are being managed.

My only instinct is to use the width prop on columns that might change size so they are always wide enough to fit the longest content (in our case) - specifically, that means 'servers', 'status', and 'location' i think. Not sure if that would work?

**Review by mmenestr — APPROVED** [review]

**Review by jenny-s51 — COMMENTED** [review]
Thank you @nicolethoen for your awesome suggestions!! The additional documentation will definitely be super helpful for consumers and the static column widths were a great idea 🙌

File: packages/react-table/src/docs/demos/table-demos/ColumnManagement.jsx [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+        return 'Status';
+      case 'check6':
+        return 'Location';
+      case 'check7':
+        return 'Last Modified';
+      case 'check8':
+        return 'URL';
+    }
+  };
+  matchSelectedColumnNameToAttr = name => {
```
Comment (thatblindgeye) [inline packages/react-table/src/docs/demos/table-demos/ColumnManagement.jsx:68]:
```suggestion
  const matchSelectedColumnNameToAttr = name => {
```

File: packages/react-table/src/docs/demos/table-demos/ColumnManagement.jsx [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+
+      // Remove the attributes (i.e. "columns") that were not selected
+      managedRows.forEach(item => filteredRows.push(removePropFromObject(item, updatedFilters)));
+
+      setFilters(updatedFilters);
+      setFilteredColumns(filteredColumns);
+      setFilteredRows(filteredRows);
+    }
+  };
+  unfilterAllData = () => {
```
Comment (thatblindgeye) [inline packages/react-table/src/docs/demos/table-demos/ColumnManagement.jsx:158]:
```suggestion
  const unfilterAllData = () => {
```

File: packages/react-table/src/docs/demos/table-demos/ColumnManagement.jsx [hunk 3]
```diff
@@ -0,0 +1,486 @@
...
+    setFilters([]);
+    setFilteredColumns(defaultColumns);
+    setFilteredRows(defaultRows);
+  };
+
+  handleChange = (checked, event) => {
```
Comment (thatblindgeye) [inline packages/react-table/src/docs/demos/table-demos/ColumnManagement.jsx:164]:
```suggestion
  const handleChange = (checked, event) => {
```

File: packages/react-table/src/docs/demos/table-demos/ColumnManagement.jsx [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+    setPaginatedRows(filteredRows);
+    setIsModalOpen(!isModalOpen);
+  };
+
+  const selectAllColumns = () => {
+    unfilterAllData();
+    setCheckedState(Array(columns.length).fill(true));
+  };
+
+  renderModal = () => {
```
Comment (thatblindgeye) [inline packages/react-table/src/docs/demos/table-demos/ColumnManagement.jsx:193]:
```suggestion
  const renderModal = () => {
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Awesome work on this! 🎉 I just had a few really quick changes below. Also had the following comments:

- When viewing the demo, two instances of "Loading..." text are being rendered:
    Double loading text in demo

    If I stop the page load in time, this is what it looks like in my dev tools:

    double loading text dev tools

    Are you by chance seeing this at all?
- If this has already been brought up just let me know (pretty sure this topic was mentioned somewhat recently, but can't place it..), but the column widths end up truncating when there may not be a need to. If I update the table so only the URL column is visible, it still truncates despite there being plenty of space. Not necessarily a blocker, though, since it could be out of scope/something that would more be up to the consumer to handle.

**Comment by nicolethoen** [comment]
@thatblindgeye @jenny-s51 The issue with an extra `Loading...` has been reported :)
https://github.com/patternfly/patternfly-org/issues/3209

**Review by jenny-s51 — COMMENTED** [review]
Thank you for your review and feedback @thatblindgeye ! Applied your code suggestions 👍

**Review by nicolethoen — APPROVED** [review]

**Review by thatblindgeye — APPROVED** [review]
Looks good! Awesome work again on this demo!

**Review by tlabaj — APPROVED** [review]
LGTM

**Pull request merged**